### DISCUSSION TITLE
Getting additional reports

### DISCUSSION THREAD

**Discussion by santhosh-r-urs** [pact-foundation/pact-js#1246] [Q&A]
Hi, 

How do I get additional reports apart from the one logged in the console ? 
Also, I run multiple contracts verifications, can I configure tests to receive a consolidated report ?

**Comment by portyu9** [comment]
Hi @santhosh-r-urs 

If you only need the Pact JS verifier output written to a file, current Pact JS supports `logFile`:

```js
const { Verifier } = require('@pact-foundation/pact')

await new Verifier({
    providerBaseUrl: 'http://localhost:8080',
    pactUrls: [
        './pacts/consumer-a-provider.json',
        './pacts/consumer-b-provider.json'
    ],
    logFile: './reports/pact-verification.log'
}).verifyProvider()
```

That is a **log file**, though, not a structured test report.

For JSON/JUnit/HTML reporting, the current supported option is the standalone Pact verifier CLI. You can make it available to a Node project with:

```bash
npm install --save-dev @pact-foundation/pact-cli
```

Then verify all contracts in one run:

```bash
npx pact verifier \
  --dir ./pacts \
  --hostname localhost \
  --port 8080 \
  --json ./reports/pact-verification.json \
  --junit ./reports/pact-verification.xml \
  --html ./reports/pact-verification.html
```

`--file` and `--dir` can be repeated, so multiple contracts can be verified in the **same invocation** and written to one consolidated report.

For example:

```bash
npx pact verifier \
  --file ./pacts/consumer-a-provider.json \
  --file ./pacts/consumer-b-provider.json \
  --json ./reports/all-pacts.json
```

If the pacts come from a Pact Broker, one verifier invocation can also retrieve and verify all selected consumer contracts for the provider.

One limitation: if your current Pact JS setup relies on JavaScript `stateHandlers`, those callbacks are provided by Pact JS's internal verification proxy. When moving verification to the standalone CLI, expose the equivalent provider-state setup endpoint and configure the CLI to call it.

So the current distinction is:

- `Verifier({ logFile })` → save normal Pact JS verification logs
- `pact verifier --json/--junit/--html` → structured/consolidated verification reports

Please mark this answer as accepted if it helped, thank you.
